modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -670,6 +671,26 @@ struct CV_EXPORTS_W HOGDescriptor
     void groupRectangles(std::vector<cv::Rect>& rectList, std::vector<double>& weights, int groupThreshold, double eps) const;
 };
 
+
+class CV_EXPORTS_W QRCodeEncoder {
+public:
+    enum {
+        CORRECT_LEVEL_L = 0,
+        CORRECT_LEVEL_M = 1,
+        CORRECT_LEVEL_Q = 2,
+        CORRECT_LEVEL_H = 3} QRCodeCorrectionLevel;
+    std::vector<Mat> generate(cv::String input_string, int mode, int version, int correction_level,int m, int eci,int s);
+    Mat generate(cv::String input_string, int mode, int version , int correction_level ,int m  , int eci);
```

Comment:
please use standard OpenCV's architecture API and use current names:
```suggestion
    bool generate(cv::String input, cv::OutputArray output, int version, int correction_level = cv::CORRECT_LEVEL_L, int mode = cv::QR_MODE_AUTO, int structure_number = 1, Size output_size = (23, 23));
```

`int m` - calculated parameter
`int eci ` - use only UTF-8

Reply:
`version` parameter should also have default option (in this case we calculate the smallest possible version depending on the length of the input string)